107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java

```diff
@@ -0,0 +1,139 @@
+          builder = templateBuilder;
+          log.debug("Configuration of cache {} will be supplemented by template {}", cacheName, templateName);
+        }
+      }
+
+      if (builder.hasDefaultExpiry()) {
+        useEhcacheExpiry = false;
+        builder = builder.withExpiry(result.cacheResources.getExpiryPolicy());
+      } else {
+        log.info("Cache {} will use expiry configuration from template {}", cacheName, templateName);
+      }
+
+      OnHeapStoreServiceConfiguration onHeapStoreServiceConfig = builder.getExistingServiceConfiguration(OnHeapStoreServiceConfiguration.class);
+      if (onHeapStoreServiceConfig == null) {
+        builder = builder.add(new OnHeapStoreServiceConfiguration().storeByValue(result.jsr107Configuration.isStoreByValue()));
+      }
+
+      DefaultCacheLoaderWriterConfiguration ehcacheLoaderWriterConfiguration = builder.getExistingServiceConfiguration(DefaultCacheLoaderWriterConfiguration.class);
+      if (ehcacheLoaderWriterConfiguration == null) {
+        useEhcacheLoaderWriter = false;
+        // No template loader/writer - let's activate the JSR-107 one if any
+        if (result.jsr107Configuration.isReadThrough() || result.jsr107Configuration.isWriteThrough()) {
+          CacheLoaderWriter<? super K, V> cacheLoaderWriter = result.cacheResources.getCacheLoaderWriter();
+          if (cacheLoaderWriter != null) {
+            cacheLoaderWriterFactory.registerJsr107Loader(cacheName, cacheLoaderWriter);
+          }
+        }
+      } else {
+        if (!result.jsr107Configuration.isReadThrough() && result.jsr107Configuration.isWriteThrough()) {
+          log.warn("Activating Ehcache loader/writer for JSR-107 cache {} which was neither read-through nor write-through", cacheName);
+        }
+        log.info("Cache {} will use loader/writer configuration from template {}", cacheName, templateName);
+      }
+
+      result.cacheConfiguration = builder.buildConfig(result.jsr107Configuration.getKeyType(), result.jsr107Configuration
+          .getValueType());
+
+      Eh107Expiry<K, V> expiryPolicy;
+      if (useEhcacheExpiry) {
+        expiryPolicy = new EhcacheExpiryWrapper<K, V>(result.cacheConfiguration.getExpiry());
+      } else {
+        expiryPolicy = result.cacheResources.getExpiryPolicy();
+      }
+
+      result.cacheResources = new CacheResources<K, V>(cacheName, result.cacheResources.getCacheLoaderWriter(), expiryPolicy,
+          result.cacheResources.getListenerResources());
+      result.jsr107Configuration = new Eh107CompleteConfiguration<K, V>(result.jsr107Configuration, result.cacheConfiguration, useEhcacheExpiry, useEhcacheLoaderWriter);
+      return result;
+    } catch (Throwable throwable) {
+      MultiCacheException mce = new MultiCacheException(throwable);
+      if (result.cacheResources != null) {
+        result.cacheResources.closeResources(mce);
+      }
+      throw mce;
+    }
+  }
+
+  String getCacheTemplateName(String cacheName) {
```

Comment:
Why is this logic part of this class and not part of the Jsr107Service impl?  That would mean we could eliminate the getDefaultTemplate method entirely right?